src/go/rpk/pkg/cli/cloud/cluster/secret/create.go

```diff
@@ -0,0 +1,77 @@
+package secret
+
+import (
+	"encoding/base64"
+	"strings"
+	"unicode"
+
+	v1alpha2 "buf.build/gen/go/redpandadata/dataplane/protocolbuffers/go/redpanda/api/dataplane/v1alpha2"
+	connect "connectrpc.com/connect"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/publicapi"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func NewCreateCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var secretName, secretValue string
+
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create a new secret",
+		Long:  "Create a new secret for your Redpanda cloud cluster",
+		Run: func(cmd *cobra.Command, _ []string) {
+			// Validate secretName for non-lowercase letters
+			for _, r := range secretName {
+				if !unicode.IsLower(r) && unicode.IsLetter(r) {
+					out.Die("secret name must be lowercase")
+				}
+			}
+
+			// Load profile
+			p, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "rpk unable to load config: %v", err)
+			if p.FromCloud {
+				makeRequest := func(url string) {
+					if url == "" {
+						out.Die("URL cannot be empty")
+					}
+					cl, err := publicapi.NewDataPlaneClientSet(url, p.CurrentAuth().AuthToken)
+					out.MaybeDie(err, "unable to initialize cloud client: %v", err)
+
+					// Create request to create secret
+					encodedSecretValue := base64.StdEncoding.EncodeToString([]byte(secretValue))
+					request := &v1alpha2.CreateSecretRequest{
+						Id:         strings.ToUpper(secretName),
+						SecretData: []byte(encodedSecretValue),
+						// this params is going to be remove in the future
+						Scopes: []v1alpha2.Scope{v1alpha2.Scope_SCOPE_REDPANDA_CONNECT},
+					}
+					response, err := cl.Secrets.CreateSecret(cmd.Context(), &connect.Request[v1alpha2.CreateSecretRequest]{Msg: request})
+					out.MaybeDie(err, "unable to create secret: %v", err)
+					out.Die("Secret %s created successfully", response.Msg.Secret.Id)
+				}
+
+				if p.CloudCluster.IsServerless() {
+					url := p.AdminAPI.Addresses[0]
```

Comment:
Also, please make sure that Addresses is non nil

Reply:
it is handling in the next line